_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,248 @@
+* I need to run the WildFly server in debug mode, so I can use IDE tools to remotely debug the code.
+
+With the use of the `wildfly-maven-plugin`, the above goals can be achieved in an automated way. Here is an example of the configuration to show the use of the plugin to achieve the above goals:
+
+* https://github.com/jberet/jberet-examples/blob/main/deployment/pom.xml[jberet-examples/deployment/pom.xml at main · jberet/jberet-examples]
+
+The `wildfly-maven-plugin` configuration in the above `pom.xml` looks like this:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>${version.maven.wildfly.plugin}</version>
+    <configuration>
+        <galleon-options>
+            <jboss-fork-embedded>true</jboss-fork-embedded>
+        </galleon-options>
+    </configuration>
+    <executions>
+        <execution>
+            <id>server-provisioning</id>
+            <phase>generate-test-resources</phase>
+            <goals>
+                <goal>provision</goal>
+            </goals>
+            <configuration>
+                <provisioning-dir>${jboss.home}</provisioning-dir>
+                <galleon-options>
+                    <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                </galleon-options>
+                <feature-packs>
+                    <feature-pack>
+                        <groupId>${server.test.feature.pack.groupId}</groupId>
+                        <artifactId>${server.test.feature.pack.artifactId}</artifactId>
+                        <version>${wildfly.version}</version>
+                        <inherit-configs>false</inherit-configs>
+                        <included-configs>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone-full.xml</name>
+                            </config>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone.xml</name>
+                            </config>
+                        </included-configs>
+                        <excluded-packages>
+                            <name>docs.schema</name>
+                            <name>appclient</name>
+                            <name>domain</name>
+                        </excluded-packages>
+                    </feature-pack>
+                </feature-packs>
+                <channels>
+                    <channel>
+                        <manifest>
+                            <groupId>org.jberet</groupId>
+                            <artifactId>jberet-channel-manifest</artifactId>
+                            <version>${version.jberet}</version>
+                        </manifest>
+                    </channel>
+                </channels>
```

Comment:
This configuration should all be moved above. The reason being if it's in the default configuration, it can be used from other goals specified on the command. For example `wildfly:dev`.

Reply:
Thanks for the suggestion James! I have created a PR to move the configuration in the project: https://github.com/jberet/jberet-examples/pull/6